src/Handlers/ResultHandler.php

```diff
@@ -95,61 +96,43 @@ public function __construct(
      * @param AsyncPaymentTransactionStruct $transaction
      * @param Request $request
      * @param SalesChannelContext $salesChannelContext
-     * @throws PaymentCancelledException
      * @throws PaymentFailedException
+     * @throws PaymentCancelledException
      */
     public function processResult(
         AsyncPaymentTransactionStruct $transaction,
         Request $request,
         SalesChannelContext $salesChannelContext
     ) {
-        // Retrieve paymentResponse and if it is
-        $paymentResponse = $this->paymentResponseService->getWithOrderTransaction(
-            $transaction->getOrderTransaction()
-        );
-
-        $result = $this->paymentResponseHandlerResult->createFromPaymentResponse(
-            $paymentResponse
-        );
+        // Retrieve paymentResponse and if it exists
+        $paymentResponse = $this->paymentResponseService->getWithOrderTransaction($transaction->getOrderTransaction());
 
-        if ('RedirectShopper' === $result->getResultCode()) {
-            // Validate 3DS1 parameters
-            // Get MD and PaRes to be validated
-            $md = $request->request->get('MD');
-            $paRes = $request->request->get('PaRes');
-            $redirectResult = $request->query->get(self::REDIRECT_RESULT);
-            $payload = $request->query->get(self::PAYLOAD);
-
-            $details = [];
-
-            // Construct the details object for the paymentDetails request
-            if (!empty($md)) {
-                $details[self::MD] = $md;
-            }
+        if (!$paymentResponse) {
+            throw new PaymentFailedException('Payment response not found.');
+        }
 
-            if (!empty($paRes)) {
-                $details[self::PA_RES] = $paRes;
-            }
+        $result = $this->paymentResponseHandlerResult->createFromPaymentResponse($paymentResponse);
 
-            if (!empty($redirectResult)) {
-                $details[self::REDIRECT_RESULT] = $redirectResult;
-            }
-
-            if (!empty($payload)) {
-                $details[self::PAYLOAD] = $payload;
-            }
+        if ('RedirectShopper' === $result->getResultCode()) {
+            $redirectResult = $request->query->get(
+                self::REDIRECT_RESULT,
```

Comment:
is it always only self::REDIRECT_RESULT? (at least for the current payment methods that we do support)

Reply:
Yes, `payload` was replaced in [v66](https://docs.adyen.com/online-payments/release-notes?tab=npm-recommended-_1666a37f_1#checkout-api-v66) for 
> redirects, except 3D Secure redirects

and then in [v67](https://docs.adyen.com/online-payments/release-notes?tab=script_1666a37f_2#checkout-api-v67):
>For redirect 3D Secure authentication, you now need to submit the redirectResult appended to the returnUrl when the shopper is redirected back to you. This replaces the `MD` and the `PaRes` parameters used in previous versions, and aligns the flow for 3D Secure redirects with all other redirects.